sdk/identity/azure-identity/src/main/java/com/azure/identity/DefaultAzureCredentialBuilder.java

```diff
@@ -197,4 +218,20 @@ private ArrayList<TokenCredential> getCredentialsChain() {
         output.add(new AzurePowerShellCredential(tenantId, identityClientOptions.clone()));
         return output;
     }
+
+    private WorkloadIdentityCredential getWorkloadIdentityCredentialIfAvailable() {
+        Configuration configuration = identityClientOptions.getConfiguration() == null
+            ? Configuration.getGlobalConfiguration().clone() : identityClientOptions.getConfiguration();
+
+        String tenantId = configuration.get(Configuration.PROPERTY_AZURE_TENANT_ID);
+        String federatedTokenFilePath = configuration.get(AZURE_FEDERATED_TOKEN_FILE);
+        String azureAuthorityHost = configuration.get(Configuration.PROPERTY_AZURE_AUTHORITY_HOST);
+        if (!(CoreUtils.isNullOrEmpty(tenantId)
+            || CoreUtils.isNullOrEmpty(federatedTokenFilePath)
+            || (CoreUtils.isNullOrEmpty(managedIdentityClientId) && CoreUtils.isNullOrEmpty(workloadIdentityClientId))
+            || CoreUtils.isNullOrEmpty(azureAuthorityHost))) {
+            return new WorkloadIdentityCredential(managedIdentityClientId, tenantId, federatedTokenFilePath, identityClientOptions.setAuthorityHost(azureAuthorityHost).clone());
```

Comment:
Forgive me if I'm just failing to parse Java here, but is the value of `workloadIdentityClientId` ever used?

Reply:
yeah, that logic is handled behind the scenes, that compromises readability here.
I've changed the logic to handle it here and make it more intuitive and readable.